cmd/agent.go

```diff
@@ -69,7 +74,17 @@ func (ar AgentRunner) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	defer ar.closePluginClients()
```

Comment:
This line should not be removed. It kills all of the plugins, once the process is over at the end of the method.
If it is not called, the plugins remain open, and the next run will add another set and so forth until the machine is exhausted with processes. 
Let's work together to see why this is creating an issue.